base/strings/util.jl

```diff
@@ -388,13 +388,83 @@ _replace(io, repl, str, r, pattern) = print(io, repl)
 _replace(io, repl::Function, str, r, pattern) =
     print(io, repl(SubString(str, first(r), last(r))))
 
-replace(str::String, pat_repl::Pair{Char}; count::Integer=typemax(Int)) =
-    replace(str, equalto(first(pat_repl)) => last(pat_repl); count=count)
-replace(str::String, pat_repl::Pair{<:Union{Tuple{Vararg{Char}},AbstractVector{Char},Set{Char}}};
-        count::Integer=typemax(Int)) =
-    replace(str, occursin(first(pat_repl)) => last(pat_repl), count)
+# replace Char with function that compares with this Char
+pairmap(p::Pair{Char}) = equalto(p.first) => p.second
+
+# replace collection of Char with function that checks occurrence in this collection
+function pairmap(p::Pair{<:Union{Tuple{Vararg{Char}},AbstractVector{Char},Set{Char}}})
+    occursin(p.first) => p.second
+end
+pairmap(p::Pair) = p
+
+# find the first of a list of patterns.
+# if several match at the same start position prefer the longer one
+# if also the length is equal, prefer the first in list
+# return value of the successful search and the pair
```

Comment:
"value"  -> "index range"

Also, better not add a blank line after the comment so that it's obvious it's attached to the function. Same below.

Reply:
done!